service/src/java/org/apache/hive/service/cli/operation/OperationManager.java

```diff
@@ -201,11 +202,28 @@ private void addOperation(Operation operation) {
     }
   }
 
+  public void updateQueryTag(String queryId, String queryTag) {
+    Operation operation = queryIdOperation.get(queryId);
+    if (operation != null) {
+      if (queryTagToIdMap.get(queryTag) != null) {
```

Comment:
Shall use containsKey() method instead of get().

Reply:
i think get it safer in case somebody put a  queryTag as null